client/state/media/actions/index.js

```diff
@@ -0,0 +1,2 @@
+export * from 'state/media/actions/sync';
+export * from 'state/media/actions/async';
```

Comment:
Hmm, you're right. Mostly I just wanted to avoid turning `state/media/actions` into a several hundred lines long file; I'd almost just prefer putting each into their own file, though that creates more export-related boilerplate.

Reply:
I didn't see a convention like this before, but it looks like a good idea 🙂

Reply:
So, as we're going forward is media file uploads something which could be handled on the data layer @jsnajdr ?

Reply:
Yes, eventually, all media requests, both fetches and modifications, should be handled by the data layer.

Reply:
With data layer, all the action creators would be trivial, i.e., only creating the action object,  and they wouldn't contain any thunks with code. That code would move to the data layer handlers.